helios-client/src/main/java/com/spotify/helios/common/descriptors/HttpHealthCheck.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.common.descriptors;
+
+import com.google.common.base.Objects;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import org.jetbrains.annotations.Nullable;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+public class HttpHealthCheck extends HealthCheck {
+
+  private final String path;
+  private final String port;
+
+  public HttpHealthCheck(@JsonProperty("path") @Nullable final String path,
+                         @JsonProperty("port") @Nullable final String port) {
```

Comment:
I think Nullable is used to indicate that a null value is valid and expected, which I don't think it is in this case? There are other parts of the code which assume the getters will return a non-null value. Right now those generate findbugs warnings since they are Nullable. So I think we'd either need to handle nulls in those cases, or make this not nullable.